backend/docker.go

```diff
@@ -295,25 +295,22 @@ func buildDockerImageSelector(selectorType string, client *docker.Client, cfg *c
 	}
 }
 
-func dockerImageIDNameFromSelection(selection string) (string, string) {
-	parts := strings.SplitN(strings.TrimSpace(selection), ";", 2)
-	if len(parts) == 2 {
-		return parts[0], parts[1]
-	}
-	return parts[0], parts[0]
-}
-
-func (p *dockerProvider) dockerImageIDFromName(ctx gocontext.Context, imageName string) string {
+// dockerImageNameForID returns a human-readable name for the image with the requested ID.
+// Currently, we are using the tag that includes the stack-name (e.g "travisci/ci-garnet:packer-1505167479") and reverting back to the ID if nothing is found.
+func (p *dockerProvider) dockerImageNameForID(ctx gocontext.Context, imageID string) string {
 	images, err := p.client.ImageList(ctx, dockertypes.ImageListOptions{All: true})
 	if err != nil {
-		return imageName
+		return imageID
 	}
-
-	imageID, _, err := findDockerImageByTag([]string{imageName}, images)
-	if err != nil {
-		return imageName
+	for _, image := range images {
+		if image.ID == imageID {
+			for _, tag := range image.RepoTags {
+				if strings.HasPrefix(tag, "travisci/ci-") {
```

Comment:
Maybe make this configurable with `"travisci/ci-"` as the default?  I'm thinking of installations that use custom images...

Reply:
@meatballhat Good point, I didn't think of that. 

How about using the RepoDigest instead (so it will be something like `travisci/ci-garnet@sha256:c759b5a71f8efec6c83e5271584e4bc8345eb6804ab0fd57a293772123edec24`) or just including all the tags in the name?

Reply:
@bogdanap I think I prefer the `travisci/ci-{stack}:{tag}` format instead of RepoDigest 👍

Reply:
Ok, `travisci/ci-{stack}:{tag}` it is then.
Although I looked into adding it as a config option and it feels a bit unnecessary, especially since the worker has a lot of those already. 
I am inclined to think that having the image-id as a default here is good enough for identification, but am happy to hear opinions to the contrary.